buildbot_nix/nix_status_generator.py

```diff
@@ -131,26 +190,45 @@ async def generate(
                 formatter, master, reporter, data
             )
 
-            if event in {"started-nix-eval", "finished-nix-eval"}:
-                report["builds"][0]["properties"]["status_name"] = (
-                    "nix-eval",
-                    "generator",
-                )
-            if (event in {"started-nix-build", "finished-nix-build"}) and report[
-                "builds"
-            ][0]["properties"]["status_name"][0] == "nix-eval":
-                report["builds"][0]["properties"]["status_name"] = (
-                    "nix-build",
-                    "generator",
-                )
-            if event in {"finished-nix-eval", "finished-nix-build"}:
-                report["builds"][0]["complete"] = True
-                report["builds"][0]["complete_at"] = datetime.now(tz=UTC)
+            event_typed: CombinedBuildEvent = CombinedBuildEvent(event)
+            match event_typed:
+                case (
+                    CombinedBuildEvent.STARTED_NIX_EVAL
+                    | CombinedBuildEvent.FINISHED_NIX_EVAL
```

Comment:
Oh, this might be actually the wrong syntax.

Reply:
Ive no clue how `match` works in python so cant comment